packages/calling-stateful-client/src/IncomingCallDeclarative.ts

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { IncomingCall } from '@azure/communication-calling';
+import { CallContext } from './CallContext';
+
+/**
+ * @beta
+ * Proxies the {@link @azure/communication-calling#IncomingCall} interface.
+ */
+export type DeclarativeIncomingCall = IncomingCall;
+
+/**
+ * @private
+ */
+export class ProxyIncomingCall implements ProxyHandler<DeclarativeIncomingCall> {
+  private _context: CallContext;
+
+  constructor(context: CallContext) {
+    this._context = context;
+  }
+
+  public get<P extends keyof IncomingCall>(target: IncomingCall, prop: P): any {
+    switch (prop) {
+      case 'accept': {
+        return this._context.withAsyncErrorTeedToState(async function (...args: Parameters<IncomingCall['accept']>) {
+          return await target.accept(...args);
+        }, 'IncomingCall.accept');
+      }
+      case 'reject': {
+        return this._context.withAsyncErrorTeedToState(async function (...args: Parameters<IncomingCall['reject']>) {
+          return await target.reject(...args);
+        }, 'IncomingCall.reject');
+      }
+      default:
+        return Reflect.get(target, prop);
```

Comment:
What does Reflect do here?

Reply:
If we don't proxy a function on incomingCallObject, it exposes the default implementation